packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -161,7 +161,7 @@ module.exports = async (api, args = {}, pluginSource) =>
       },
       (err, results) => {
         if (err) {
-          reporter.error(`${pluginName} returned an error`, err)
```

Comment:
Since `panicOnBuild` doesn't exist, you can do the same thing by adding a conditional `if (process.env.NODE_ENV === "production") { /* panic */ } else { /* error */ }`